### DISCUSSION TITLE
Evaluation Step Doesn't Use GPU?

### DISCUSSION THREAD

**Discussion by fresh4** [coqui-ai/TTS#2430] [General Q&A]
I don't know if this is a config thing or program thing; I obviously have other issues because I keep getting warnings while synthesizing the test sentences:

`Decoder stopped with \`max_decoder_steps\``

about 5 times and each time it takes about like 5+ minutes, which is significantly slower than the actual training itself. (The test sentences in tensorboard come out to like 13 minutes somehow)

Anyway, in the assumption that further training might train this out, I've noticed that during this evaluation step, the GPU is not being utilized. It hits 100% during training but not during this step. I think it might run much faster if this step is loaded onto the GPU, so does anyone know if that's possible, a config setting, or where that might be modified?

It might not even be feasible or built for that, idk, but I haven't seen any information on it, so I thought I'd ask.

**Comment by p0p4k** [comment]
What is your `dataset size` and `eval_size` respectively? Then tell me your `eval_batch_size` as well.

**Comment by p0p4k** [comment]
Decoder stopped with \max_decoder_steps`` This is just to stop the model from generating for long and save computation. In general a transformer based autoregressive model generates output one sequence element at a time (here the mel frame), and stops till it generates a <eos> (end of sequence) token. However, we cannot wait for the model to do this everytime during training as it might never generate an <eos> token and thus never stop the generation. To solve this issue we force the model to generate a fixed max_decoder_steps of tokens (each token being a mel spec frame with <sos> and <eos> tokens included). So do not worry about the error for now.
